### PR TITLE
Support setting basic literals for BigQuery

### PR DESCRIPTION
**Pull request by thijs-obj**
Support literals for basic types on BigQuery. E.g. allow things like `df['x'] = 123`

There is still work to do to support more types, but this supports the basic types

### CODE REVIEW

File: bach/tests/functional/bach/test_df_setitem.py [hunk 1]
```diff
@@ -47,9 +48,23 @@ def test_set_const_int(engine):
     check_set_const(engine, 2147483648, SeriesInt64, 'bigint')
 
 
-def test_set_const_float():
-    engine = get_postgres_engine_dialect().engine  # TODO: BigQuery
+def test_set_const_float(engine):
     check_set_const(engine, 5.1, SeriesFloat64, 'double precision')
+    check_set_const(engine, 0.0, SeriesFloat64, 'double precision')
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_df_setitem.py:53]: These are some interesting cases for which we previously didn't have a test

File: bach/bach/series/series_datetime.py [hunk 2]
```diff
@@ -313,12 +318,13 @@ class SeriesTimedelta(SeriesAbstractDateTime):
     supported_value_types = (datetime.timedelta, numpy.timedelta64, str)
 
     @classmethod
-    def supported_literal_to_expression(cls, literal: Expression) -> Expression:
+    def supported_literal_to_expression(cls, dialect: Dialect, literal: Expression) -> Expression:
         return Expression.construct('cast({} as interval)', literal)
```
Comment (KathiaBarahona) [inline bach/bach/series/series_datetime.py:322]: why not? I'm aware SeriesTimedelta is supported only for Postgres, but IMHO we should still get the db type from the mapping.
```suggestion
        db_dialect = DBDialect.from_dialect(dialect)
        db_dtype = cls.supported_db_dtype[db_dialect]
        return Expression.construct(f'cast({{}} as {db_dtype})', literal)
```

File: bach/bach/series/series_numeric.py [hunk 3]
```diff
@@ -186,18 +188,23 @@ class SeriesInt64(SeriesAbstractNumeric):
...
+            # consistently get bigints, so always cast the result
+            # See the section on numeric constants in the Postgres documentation
+            # https://www.postgresql.org/docs/14/sql-syntax-lexical.html#SQL-SYNTAX-CONSTANTS
+            return Expression.construct(f'cast({{}} as {cls.get_db_dtype(dialect)})', literal)
+        if is_bigquery(dialect):
+            # BigQuery has only one integer type, so there is no confusion between 32-bit and 64-bit integers
+            # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#integer_type
+            # https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#integer_literals
+            return literal
+        raise DatabaseNotSupportedException(dialect)
```
Comment (KathiaBarahona) [inline bach/bach/series/series_numeric.py:204]: shouldn't we raise also this exception for other Series that don't support BigQuery? Such as `SeriesTimeDelta` and `SeriesJsonb`

File: bach/bach/series/series_datetime.py (hunk 2 above)
Comment (thijs-obj) [reply bach/bach/series/series_datetime.py]: Changed in 904c7a5f330f9d59c1ec1a849d48b9040ffdd945

File: bach/bach/series/series_numeric.py (hunk 3 above)
Comment (thijs-obj) [reply bach/bach/series/series_numeric.py:204]: Changed in 904c7a5f330f9d59c1ec1a849d48b9040ffdd945, we now raise that exception in not supported series for BigQuery

**Review by KathiaBarahona — APPROVED** [review]
lgtm

**Pull request merged**